### ISSUE TITLE
challenge_required everytime

### ISSUE THREAD

**Issue by chavezjose** [LevPasha/Instagram-API-python#663]
> Request return 400 error!
{'message': 'challenge_required', 'challenge': {'url': 'https://i.instagram.com/challenge/****/*******/', 'api_path': '/challenge/****/*******/', 'hide_webview_header': True, 'lock': True, 'logout': False, 'native_flow': True}, 'status': 'fail', 'error_type': 'checkpoint_challenge_required'}
Can't login!

Getting this every single time, even after following the URL and verifying. Used two accounts both tied to a phone number (even though it's the same #). Just SOL at this point or what?

**Comment by MarcosChabolla** [comment]
I'm having the same issue. Unable to get past this. 
Any luck?

**Comment by chavezjose** [comment]
Nope. Was only trying it for a small project, but didn't have enough interest to keep tinkering.

**Comment by SeWizarD** [comment]
I guess Instagram needs more Data like a legit fingerprint (maybe we're sending an old one [I actually don't know])

**Comment by arijlopez** [comment]
Yeah, it seems that Instagram is getting more strict with accounts that are using bots. I m trying to find a workaround without luck so far

**Comment by fjkl446** [comment]
hey guys hope this will help you


`import json
from InstagramAPI import InstagramAPI

api = InstagramAPI(YOURID, YOURPW)
api.login()

print(api.LastJson)
challenge_message = api.s.get(api.API_URL + api.LastJson['challenge']['api_path'][1:])
print(json.loads(challenge_message.text))

challenge_choice = api.s.post(api.API_URL + api.LastJson['challenge']['api_path'][1:], data={'choice': '1'})
print(json.loads(challenge_choice.text))

input_code = input('enter code')
print(input_code)

a = api.s.post(api.API_URL + api.LastJson['challenge']['api_path'][1:], data={'security_code': int(input_code)})
print(json.loads(a.text))

api.login()`


remember to enter you code from your email 
✌

**Comment by rockboynton** [comment]
> remember to enter you code from your email

What code?

**Comment by saifullah9002** [comment]
i get login success message but i am unable to upload any media and i get no error message

**Comment by sahiltspaceo** [comment]
Try this
https://github.com/LevPasha/Instagram-API-python/issues/706#issuecomment-560248053

**Comment by orest1114** [comment]
Fix
#718
